nexpose/nexpose.py

```diff
@@ -2052,6 +2062,33 @@ def GenerateScanReport(self, scan_or_id):
         body = ''.join(data[4:-1])
         return as_xml(base64.urlsafe_b64decode(body))
 
+    def GenerateAdHocReport(self, adhoc_report_configuration):
+        """
+        Generate adhoc report and return decoded contents.
+        """
+        # TODO: add optional filename param to store the report to disk instead of memory
+        self._RequireInstanceOf(adhoc_report_configuration, AdhocReportConfiguration)
+        data = self.RequestAdhocReport(adhoc_report_configuration.AsXML())
+        data = self.VerifySuccess(data)
+        # TODO: figure out a way to handle this safely/correctly for different formats
+        data = data.tail.replace('\r', '').strip().split('\n')
+        # assert data[1] == 'Content-Type: text/xml; name=report.xml'
+        assert data[2] == 'Content-Transfer-Encoding: base64'
+        assert data[3] == ''
+        assert data[0] == data[-1][:-2]
+        body = ''.join(data[4:-1])
+        return base64.urlsafe_b64decode(body)
+
+    def SaveReportConfiguration(self, report_configuration, generate_now=False):
+        """
+        Save the configuration of a report and return the id of the saved report config.
+        If successful, the id will also have been updated in the provided ReportConfiguration object.
+        To create a new report, specify -1 as id.
+        """
+        # TODO: how to pass generate_now param downstream?
```

Comment:
As in how to pass to Nexpose to generate the report now? If so: 
`<ReportSaveRequest session-id="{sid}" generate-now="{1/0}">`

Reply:
The problem is the magic under the hood with the `ExecuteSave()` function, it takes in a function as an argument and no way to pass arguments to that function. I suppose I could do something with `**args` or `**kwargs` to keep it compatible elsewhere. After this PR is done it's probably time to rewrite all of that stuff to just use the requests lib instead and reduce some of the "magic" involved.